airbyte-lib/airbyte_lib/telemetry.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+import datetime
+import os
+from dataclasses import asdict, dataclass
+from enum import Enum
+from typing import Any, Dict, Optional
+
+import requests
+from airbyte_lib.version import get_version
+
+# TODO: Use production tracking key
+TRACKING_KEY = "jxT1qP9WEKwR3vtKMwP9qKhfQEGFtIM1" or str(os.environ.get("AIRBYTE_TRACKING_KEY"))
+
+
+class SourceType(str, Enum):
+    VENV = "venv"
+    LOCAL_INSTALL = "local_install"
+
+
+class CacheType(str, Enum):
+    # TODO: Extend once proper caches have been merged
+    IN_MEMORY = "in_memory"
+    STREAMING = "streaming"
+
+
+class SyncState(str, Enum):
+    STARTED = "started"
+    FAILED = "failed"
+    SUCCEEDED = "succeeded"
+
+
+@dataclass
+class SourceTelemetryInfo:
+    name: str
+    type: SourceType
+    version: Optional[str]
+
+
+def send_telemetry(source: SourceTelemetryInfo, cache_type: CacheType, state: SyncState, number_of_records: Optional[int] = None):
+    # If DO_NOT_TRACK is set, we don't send any telemetry
+    if os.environ.get("DO_NOT_TRACK"):
+        return
+
+    current_time = datetime.datetime.utcnow().isoformat()
+    payload: Dict[str, Any] = {
+        "anonymousId": "airbyte-lib-user",
+        "event": "sync",
+        "properties": {
```

Comment:
I think I'd like to request one more attribute here, which is capturing the presence of a "CI" flag in os.environ. 

CI tests are very noisy on our metrics - with the possibility to make up 50-80% of usage stats in some extreme cases, or in specific cross-sections of usage. Luckily, the industry has a convention:

https://chat.openai.com/share/eb500944-8bb6-43a1-b64e-5f901f4587c4

_(Turns out "CI environment variable" is hard to google. I've previously seen a third party site advocating for this standard but I couldn't find it today so the above link is a ChatGPT conversation.)_

Rather than return the value stored in the "CI" environment variable, I think it is sufficient to list it in 'extra_flags' or similar, or just as its own boolean ("True" means present").

Wdyt?